### PR TITLE
Move archive info into dropdown

### LINKED ISSUE #235: Show archive info in modal
### Changes
- Show archive info in modal dialog, similar to RWP info modal
- Add location bar dropdown menu item to open archive info

### Mockups
[Archive Info Dialog Popup](https://www.figma.com/file/M5ZLsZNgK15RWzZF1SeoTM/ReplayWeb.Page-2.0-UI?type=design&node-id=645%3A8116&mode=design&t=gnrTUGCxNsvMttbx-1)

### PR DESCRIPTION
**Pull request by emma-sg**
Closes #235 

## Summary

Moves the archive info from a tab into a dialog accessible via the dropdown menu alongside the address bar.

## Details

The dialog was implemented with [`sl-dialog`](https://shoelace.style/components/dialog).

Design moves closer to the 2.0 design, though not all the way — see attached screenshot. 
- Copy to clipboard buttons are now always visible, and show up on more fields
- Fields use a monospace treatment
- Item ID was moved up
- Description fields were removed, as per discussion with @Shrinks99 

I also:
- Improved some of the types, specifically doing better nullish checks, which also resulted in a couple additions to the shared `Item` type.
- Did some renaming of "coll" to "item" in the affected parts of the codebase
- Swapped the custom copy button to Shoelace's

## Screenshots

Screenshot 2023-11-02 at 5 26 25 PM

Screenshot 2023-11-06 at 6 27 13 PM

## Testing
Changes were tested by hand locally using various example WACZ files.

### CODE REVIEW

File: src/coll.ts [hunk 1]
```diff
@@ -149,6 +154,8 @@ class Coll extends LitElement {
 
   private observer?: IntersectionObserver;
 
+  private archiveInfoDialog: Ref<SlDialog> = createRef();
```
Comment (ikreymer) [inline src/coll.ts:157]: Didn't know about this lit function! Here its mostly used to just show/hide the dialog, right?
I guess another approach that we've used is having a separate bool that toggles if the dialog is rendered or not, but this is useful for referring to the element directly for other reasons!

Comment (emma-sg) [reply src/coll.ts:157]: Yep! In my mind this is more straightforward than keeping track of state ourselves here, which would also involve listening for a dialog close event. We don't actually need to know if the dialog is open or not here!

**Comment by ikreymer** [comment]
Looking good overall! One thing I noticed is that this also affects the home page, which uses the same component currently:

Screenshot 2023-11-03 at 11 24 40 AM

Also a design questions for @Shrinks99 how we want to handle the home page index, if we want to keep it same as is, or do something different? Fine with keeping it as is.

File: src/coll.ts (hunk 1 above)
Comment (ikreymer) [reply src/coll.ts:157]: Yeah, I guess a pattern we've used before is something like:
```
this.isDialogOpen ? html`<sl-dialog @sl-hide=${() => this.isDialogOpen = false}> ... <sl-button @click="${() => this.isDialogOpen = false}"> ... </sl-dialog>` : ''`
```
I guess either way a member variable is maintained, though only a bool in this case, and yes, need to catch all the places dialog can be closed. Don't have a strong preference either way.

**Comment by emma-sg** [comment]
Oh good catch @ikreymer! I think I might try to refactor this a little, it seems a little unnecessary that both the listing view and the details view use the same component just because they both display some of the same information.

**Comment by Shrinks99** [comment]
> Also a design questions for @Shrinks99 how we want to handle the home page index, if we want to keep it same as is, or do something different? Fine with keeping it as is for now.

Home page is something that we haven't mocked up yet!  I imagine this will be similar to the archived items list?  Not 100% clear on how this interacts with ArchiveWeb.page, should probably clarify a few things with you before I make mockups... It's largely something I put off thinking about and figured we'd get back to it when relevant, perhaps that time is now!

**Comment by Shrinks99** [comment]
Nice!

I saw we're using Shoelace 2.8 in ReplayWeb.page, could maybe switch the copy buttons to use [`<sl-copy-button>`](https://shoelace.style/components/copy-button)??  Shoelace says they're still marked as "experimental", but I don't see this component undergoing a lot of changes any time soon....

In the app we use `btrix-copy-button` but that's just because Shoelace didn't have one yet.  I imagine we'll want to switch our component over to that once we update Shoelace over there too.

+1 for hiding collection ID on the homepage.

**Comment by emma-sg** [comment]
Did some refactoring & resolved the remaining issues here, as far as I can tell!

I've updated the screenshot of the dialog in the PR description, and here's what the index view looks like:

Screenshot 2023-11-06 at 6 27 41 PM

**Review by Shrinks99 — APPROVED** [review]

**Pull request merged**